eventuals/do-all.h

```diff
@@ -67,11 +95,68 @@ struct _DoAll {
                               StoppedException()));
                   if (counter_.fetch_sub(1) == 1) {
                     // You're the last eventual so call the continuation.
-                    k_.Start(std::move(values_));
+                    std::optional<std::exception_ptr> exception =
+                        GetExceptionIfExists();
+
+                    CHECK(exception);
+                    try {
+                      std::rethrow_exception(*exception);
+                    } catch (const StoppedException&) {
+                      k_.Stop();
+                    } catch (...) {
+                      k_.Fail(std::current_exception());
+                    }
                   }
                 }));
     }
 
+    std::tuple<
+        std::conditional_t<
+            std::is_void_v<typename Eventuals_::template ValueFrom<void>>,
+            std::monostate,
+            typename Eventuals_::template ValueFrom<void>>...>
+    GetTupleOfValues() {
+      return std::apply(
+          [](auto&... value) {
+            ((CHECK(value.index() == 0)), ...);
```

Comment:
What about using less parens: `(CHECK_EQ(0, value.index(), ...);`

If that doesn't work then: 
```suggestion
            // NOTE: not using `CHECK_EQ()` here because compiler ... please fill in Artur!
            ((CHECK(value.index() == 0)), ...);
```